ui/src/shared/components/OptIn.js

```diff
@@ -0,0 +1,203 @@
+import React, {Component, PropTypes} from 'react'
+import classnames from 'classnames'
+
+// these help ensure that blur and toggle events don't both change toggle's side
+const RESET_TIMEOUT = 300
+const TOGGLE_CLICKED_TIMEOUT = 20
+const BLUR_FOCUS_GAP_TIMEOUT = 10
+
+class OptIn extends Component {
+  constructor(props) {
+    super(props)
+
+    const {customValue, fixedValue} = props
+
+    this.state = {
+      useCustomValue: customValue !== '',
+      fixedValue,
+      customValue,
+      toggleClicked: false,
+      customValueInputBlurred: false,
+      resetTimeoutID: null,
```

Comment:
i felt the same way until reviewing some code where it didn't quite make sense to use `is` for booleans. in this case, these are one time event triggers, rather than a continuous state. feels idiomatically confusing to me to use `is` here for these. i could use `was`.